app/src/main/java/org/openobservatory/ooniprobe/model/database/Result.java

```diff
@@ -221,4 +202,23 @@ public void delete(Context c) {
 		if (this.network != null)
 			this.network.delete();
 	}
+
+	public OONIDescriptor<BaseNettest> getDescriptor(Context context) {
+		try {
+			return OONITests.valueOf(test_group_name).toOONIDescriptor(context);
+		} catch (IllegalArgumentException e) {
+			return new OONIDescriptor<>(
+                    test_group_name,
+                    test_group_name,
+                    test_group_name,
+                    test_group_name,
+                    test_group_name,
+                    R.color.color_gray7_1,
+                    test_group_name,
+                    R.string.TestResults_NotAvailable,
+                    new ArrayList<>(),
+                    new ArrayList<>()
+            );
```

Comment:
The indenting here is off. Also I do not fully understand what you're doing here. Are you returning the empty base nettest in this case? What is the implication of returning it? When could the exception occur?

Reply:
I would recommend to use a more idiomatic pattern. Return an optional type and make it null in this case, then modify the caller to check whether the returned value is empty. This feels better than returning the zero value object because it's more robust in that it reduces the coupling between parts of the codebase.

Reply:
Let me do that instead. I didn't want to do so many null checks in java. 🤣

Reply:
On second thought, i dont think we should change the current way it works

Reply:
Oh, right, the consumer is still written in Java 🤦 

See, I was starting to think in Kotlin!

Reply:
Right now, It works by using the enum as a lookup. Once descriptors are saved in the database, there is still a possibility that the test name will not be found. to avoid displaying errors and blank screens, it is better to have the test name in all areas where any other test information about that test is to be displayed.